src/integrations/bluetooth-low-energy/bluetooth-low-energy.service.ts

```diff
@@ -106,18 +106,24 @@ export class BluetoothLowEnergyService extends KalmanFilterable(Object, 0.8, 15)
    * @param event - Event with new distance data
    */
   handleNewDistance(event: NewDistanceEvent): void {
-    const sensorId = makeId(`ble ${event.tagId}`);
+    const sensorId = makeId(`ble ${event.tagName}`);
     let sensor: RoomPresenceDistanceSensor;
     if (this.entitiesService.has(sensorId)) {
       sensor = this.entitiesService.get(sensorId) as RoomPresenceDistanceSensor;
     } else {
-      sensor = this.createRoomPresenceSensor(
-        sensorId,
-        event.tagId,
-        event.tagName
-      );
-    }
-
+      if (this.isOnWhitelist(event.tagId)) {
+        sensor = this.createRoomPresenceSensor(
+          sensorId,
+          event.tagId,
```

Comment:
This change could be a bit cleaner by just using ternary operator for the second parameter, no need to repeat all the rest.